packages/esm-patient-chart-app/src/visit/visit-form/visit-form.component.tsx

```diff
@@ -113,37 +116,119 @@ const StartVisitForm: React.FC<StartVisitFormProps> = ({
       {},
     );
 
-    return z.object({
-      visitStartDate: z.date(),
-      visitStartTime: z
-        .string()
-        .refine((value) => value.match(time12HourFormatRegex), t('invalidTimeFormat', 'Invalid time format')),
-      visitStartTimeFormat: z.enum(['PM', 'AM']),
-      visitStopDate: displayVisitStopDateTimeFields ? z.date() : z.date().optional(),
-      visitStopTime: displayVisitStopDateTimeFields
-        ? z
-            .string()
-            .refine((value) => value.match(time12HourFormatRegex), t('invalidTimeFormat', 'Invalid time format'))
-        : z.string().optional(),
-      visitStopTimeFormat: displayVisitStopDateTimeFields ? z.enum(['PM', 'AM']) : z.enum(['PM', 'AM']).optional(),
-      programType: z.string().optional(),
-      visitType: z.string().refine((value) => !!value, t('visitTypeRequired', 'Visit type is required')),
-      visitLocation: z.object({
-        display: z.string(),
-        uuid: z.string(),
-      }),
-      visitAttributes: z.object(visitAttributes),
-    });
-  }, [t, config, displayVisitStopDateTimeFields]);
+    let maxVisitStartDatetime: Date = null,
+      minVisitStopDatetime: Date = null;
+
+    if (visitToEdit?.encounters?.length) {
+      const allEncountersDateTime = visitToEdit?.encounters?.map(({ encounterDatetime }) =>
+        Date.parse(encounterDatetime),
+      );
+      maxVisitStartDatetime = new Date(Math.min(...allEncountersDateTime));
+      minVisitStopDatetime = new Date(Math.max(...allEncountersDateTime));
+    }
+
+    const dateTimeSchema = z
+      .object({
+        date: z.date().max(new Date()),
```

Comment:
It seems slightly risky to have multiple `new Date()` instances, i.e., both here and in the `refine()` call. It would probably be good to have them be consistent.